File: backend/src/main/java/com/smartcampus/backend/dto/weather/CurrentWeatherDTO.java
package com.smartcampus.backend.dto.weather;

import lombok.*;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class CurrentWeatherDTO {
    private Double temperature;
    private Double windspeed;
    private Integer winddirection;
    private Integer weathercode;
    private String description; // deduit du weathercode + is_day
    private String icon;
    private String time;
    private Integer isDay;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/weather/CurrentWeatherDto.kt
package com.ObservatoireCampus.mobile.model.weather

data class CurrentWeatherDto(
    val temperature: Double?,
    val windspeed: Double?,
    val winddirection: Int?,
    val weathercode: Int?,
    val description: String?,
    val icon: String?,
    val time: String?,
    val isDay: Int?
)